spec/truffle/interop/java_string_spec.rb

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+# 
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+require_relative '../../ruby/spec_helper'
+
+describe "Java strings" do
+  
+  it "are boxed" do
+    Truffle::Interop.boxed?(Truffle::Interop.to_java_string('test')).should be_true
```

Comment:
Java Strings are not boxed, they are primitives. Why does this pass?
Does the call to boxed?/IS_BOXED convert the argument? I think it should not.

Reply:
`to_java_string` returns a java.lang.String, right?

Reply:
Yes this is confusing - see my new commit.